proselint/tools.py

```diff
@@ -174,7 +169,11 @@ def lint(self, config: Config = DEFAULT) -> list[LintResult]:
                     )
                     for check in registry.get_all_enabled(config["checks"])
                     for result in check.check_with_flags(self.content)
-                    if not self.is_quoted_pos(result.start_pos)
+                    if (
+                        not self.is_quoted_pos(result.start_pos)
+                        or check.flags.allow_quotes
```

Comment:
The static condition should come first here.
```suggestion
                        check.flags.allow_quotes
                        or not self.is_quoted_pos(result.start_pos)
```